### PR TITLE
setup local and production env vars for different hosts

### PR DESCRIPTION
**Pull request by muzammal-invozone**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,3 @@
+NODE_ENV=local
+LOCAL_HOST=http://subdomain.jscms.test:3000/
+LIVE_HOST=http://jscms.test:3000/
```
Comment (isheraz) [inline .env.example]: this is the HOST
LIVE_HOST will be undefined right now.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const http = require('http');
+require('dotenv').config()
+const host = process.env.NODE_ENV === "local" ? process.env.LOCAL_HOST : process.env.LIVE_HOST;
+
+const server = http.createServer((req, res) => {
```
Comment (isheraz) [inline app.js]: Please use Expressjs for defining routes and configure CORS.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    if(req.url == '/'){
+        res.write('it is Home Page');
```
Comment (isheraz) [inline app.js]: send a JSON response. this is an API and will not be going to a page directly. there will be no home page for this.

**Review by isheraz — CHANGES_REQUESTED** [review]

File: app.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const express = require("express");
+var cors = require('cors')
+const app     = express();
+app.use(cors())
```
Comment (isheraz) [inline app.js]: Awesome, Allow CORS from localhost, and jscms.test right now. 
Use an array called whitelisted = [] to specify allowed CORS

File: app.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+require('dotenv').config()
+const host = process.env.NODE_ENV === "production" ? process.env.HOST : 'http://subdomain.jscms.test:6000/';
+const port = process.env.PORT || 6000;
+
+app.get('/', function (req, res) {
```
Comment (isheraz) [inline app.js]: use _req where the variable is not being used. Also add a eslint rule to remove or prefix with _ 'underscore' for an unused variable.

**Review by isheraz — CHANGES_REQUESTED** [review]

File: app.js (hunk 4 above)
Comment (muzammal-invozone) [reply app.js]: Added!

File: app.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+const express = require("express");
+const cors = require('cors')
+const app     = express();
+
+require('dotenv').config()
+const host = process.env.NODE_ENV === "production" ? process.env.HOST : 'http://subdomain.jscms.test:6000/';
+const port = process.env.PORT || 6000;
+
+const whitelisted = ['http://localhost:6000/', 'http://jscms.test/'];
```
Comment (isheraz) [inline app.js:9]: `http://jscms.test:6000`

File: app.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+const corsOptionsDelegate = function (req, callback) {
```
Comment (isheraz) [inline app.js:10]: Awesome. Can we use es6 standards here? 
e.g: `const corsOptionsDelegate = (req, callback) => {...}`

**Review by isheraz — CHANGES_REQUESTED** [review]

**Pull request closed without merging**